code-validation/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+//! This crate is used to validate the rust code of the `bevy` website.
+//!
+//! It is currently used to validate the rust code of the offical `bevy` book.
+//! The modules represents the folder structure of the website.
+
+mod learn {
```

Comment:
Ouch. I don't love how manual this process is.

Reply:
Yeah. We can poke at fixing this later IMO.

Reply:
Agreed. This is already a huge improvement to not having code validation at all.